gui/default/syncthing/core/syncthingController.js

```diff
@@ -912,6 +923,17 @@ angular.module('syncthing.core')
             return '';
         };
 
+        $scope.findDeviceFromShort = function (shortID) {
+            var matches = $scope.devices.filter(function (n) {
+                return n.deviceID.substr(0, 7) === shortID;
+            });
+            if (matches.length !== 1) {
+                n.deviceID.name = "Unknown";
+                return n;
+            }
+            return matches[0];
+        };
+
```

Comment:
No, because A <-> B <-> C, A will never know C's name. The way we solve the AAAAA case is we makeit return an empty string instead of AAAAA, in which case we display unknown.

Sure people will upgrade, but not everyone atthe same pace.